src/core/qgsmaplayer.h

```diff
@@ -444,6 +451,12 @@ class CORE_EXPORT QgsMapLayer : public QObject
     */
     QVariant customProperty( const QString &value, const QVariant &defaultValue = QVariant() ) const;
 
+    /** Set custom properties for layer.
+     * \param properties The custom properties to set.
```

Comment:
maybe mention that any previously set props will be gone? no reason for stating the obvious in \param.